cmd/crl-updater/main.go

```diff
@@ -86,7 +86,7 @@ type Config struct {
 		// significantly longer, since a full update cycle may consist of updating
 		// many shards with varying degrees of parallelism. This value must be
 		// strictly less than the UpdatePeriod. Defaults to 1 hour.
-		UpdateTimeout config.Duration `validate:"-"`
+		UpdateTimeout config.Duration `validate:"structonly,ltcsfield=UpdatePeriod"`
```

Comment:
I'm not sure `ltcsfield=UpdatePeriod` actually works. Elsewhere we use either `-` (omit validation) or `required`.

Reply:
See https://github.com/letsencrypt/boulder/pull/6941#discussion_r1253397029, I think Phil tried it? I admit I did not try all permutations myself.

Reply:
Okay, yep, I've confirmed this validation doesn't work. This is because config.Duration doesn't have a value itself, it's a struct that contains a value. So it can't be "less than" some other field. This validation would only work if we could put it directly on the time.Duration embedded within the config.Duration, but obviously we can't, because that's shared everywhere.

And "structonly" just means "run the validations specified here, but don't descend into the struct to run validations on its fields". The only validation tags it can be combined with, because they're the only ones that natively understand structs themselves, are "required" and "omitempty". So combining "structonly" with "ltcsfield" doesn't do anything.

I think things might be different if instead of
```go
type Duration struct {
  time.Duration `validate:"required"`
}
```
we had
```go
type Duration time.Duration
```
Then it would actually _be_ a non-composite type, rather than a struct embedding a non-composite type, and I think ltcsfield would work properly. But making that change would cause ripples across all of our configs, I suspect.

Reply:
😵

Reply:
Thanks for taking a second look.

Reply:
But this in contrast to my experience. I updated my local `updateTimeout` to be `100000s` which is far more than the `updatePeriod` of `6h`.
```
root@7e1d72baae83:/boulder# ./bin/crl-updater -config test/config-next/crl-updater.json
17:35:13.330133 6 crl-updater pOv44QY Versions: crl-updater=(Unspecified Unspecified) Golang=(go1.20.5) BuildHost=(Unspecified)
17:35:13.331237 3 crl-updater i-il5AE [AUDIT] Failed to create crl-updater: update timeout must be less than period: 27h46m40s !< 6h0m0s
```

Reply:
Samantha and I have been discussing that this PR should ship as-is and we should have a separate config validation PR for each component.